prov/sockets/src/sock_ep_msg.c

```diff
@@ -1214,6 +1305,30 @@ static struct fi_ops_cm sock_pep_cm_ops = {
 	.shutdown = fi_no_shutdown,
 };
 
+int sock_pep_getopt(fid_t fid, int level, int optname,
+		      void *optval, size_t *optlen)
+{
+	if (level != FI_OPT_ENDPOINT || optname != FI_OPT_CM_DATA_SIZE)
+		return -FI_ENOPROTOOPT;
+
+	if (*optlen < sizeof(size_t))
```

Comment:
Can you also set the optlen to sizeof(size_t) for this case (as discussed in #2089) ?